earth2studio/data/weatherbench.py

```diff
@@ -0,0 +1,241 @@
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pathlib
+import shutil
+from datetime import datetime
+
+import fsspec
+import gcsfs
+import numpy as np
+import xarray as xr
+import zarr
+from loguru import logger
+from modulus.distributed.manager import DistributedManager
+from tqdm import tqdm
+
+from earth2studio.data.utils import prep_data_inputs
+from earth2studio.lexicon import WBLexicon
+from earth2studio.utils.type import TimeArray, VariableArray
+
+LOCAL_CACHE = os.path.join(os.path.expanduser("~"), ".cache", "earth2studio")
+
+
+class WeatherBenchClimatology:
+    """
+    Climatology provided by WeatherBench2,
+    https://weatherbench2.readthedocs.io/en/latest/data-guide.html#ground-truth-datasets
+
+    |    A climatology is used for e.g. computing anomaly metrics such as the ACC.
+    |    For WeatherBench 2, the climatology was computed using a running window for
+    |    smoothing (see paper and script) for each day of year and sixth hour of day.
+    |    We have computed climatologies for 1990-2017 and 1990-2019.
+
+    Parameters
+    ----------
+    file : str, optional
+        File within gs://weatherbench2/datasets/era5-hourly-climatology/ to select,
+        by default 1990-2019_6h_1440x721.zarr
+        As of 05/03/2024 this is the following list of available files:
+            1990-2017_6h_1440x721.zarr
+            1990-2017_6h_512x256_equiangular_conservative.zarr
+            1990-2017_6h_240x121_equiangular_with_poles_conservative.zarr
+            1990-2017_6h_64x32_equiangular_conservative.zarr
+            1990-2019_6h_1440x721.zarr
+            1990-2019_6h_512x256_equiangular_conservative.zarr
+            1990-2019_6h_240x121_equiangular_with_poles_conservative.zarr
+            1990-2019_6h_64x32_equiangular_conservative.zarr
+    cache : bool, optional
+        Cache data source on local memory, by default True
+    verbose : bool, optional
+        Print download progress, by default True
+
+    Warning
+    -------
+    This is a remote data source and can potentially download a large amount of data
+    to your local machine for large requests.
+
+    Note
+    ----
+    Additional information on the data repository can be referenced here:
+
+    - https://weatherbench2.readthedocs.io/en/latest/data-guide.html#ground-truth-datasets
```

Comment:
Consider updating to https://weatherbench2.readthedocs.io/en/latest/data-guide.html#era5-climatology

Reply:
Also lets add the arxiv paper below this
https://arxiv.org/abs/2308.15560

Reply:
Good suggestions!